### PR TITLE
IPROTO-319 Fix maps with non primitive values

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.redhat.com/browse/IPROTO-319

### CODE REVIEW

File: integrationtests/src/test/java/org/infinispan/protostream/integrationtests/processor/marshaller/model/ModelWithMap.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    @ProtoField(value = 1, mapImplementation = ConcurrentHashMap.class)
+    public Map<String, Integer> getSimpleMap() {
+        return simpleMap;
+    }
+
+    public void setSimpleMap(Map<String, Integer> simpleMap) {
+        this.simpleMap = simpleMap;
+    }
+
+    // TODO: this map with UUID adapter fails to read the entries correctly
```
Comment (pruivo) [inline integrationtests/src/test/java/org/infinispan/protostream/integrationtests/processor/marshaller/model/ModelWithMap.java:24]: I would probably remove the TODO comments :+1:

**Comment by pruivo** [comment]
I tried to use a map with non-primitive keys and got this exception
```
ProtoSchemaBuilderException: The key of the map field 'uuidMap' of org.infinispan.protostream.integrationtests.processor.marshaller.model.ModelWithMap must be either a String or an integral type
```
is it a limitation of protobuf3?

**Comment by tristantarrant** [comment]
> I tried to use a map with non-primitive keys and got this exception
> is it a limitation of protobuf3?

Yes, see:

https://protobuf.dev/programming-guides/proto3/#maps

**Comment by pruivo** [comment]
merged! thanks @tristantarrant !

**Pull request merged**